### ISSUE TITLE
[fix] Add the ability for screen reader users to know that a page is the last page when navigating the Pagination component

### ISSUE THREAD

**Issue by obliviga** [trussworks/react-uswds#3093]
**ReactUSWDS Version & USWDS Version:**
```
    "@trussworks/react-uswds": "^9.1.0",
    "@uswds/uswds": "3.11.0",
```

**Describe the bug**
USWDS's [accessibility guidance](https://designsystem.digital.gov/components/pagination/#accessibility-guidance) on the Pagination component recommends the following:
```
Voice “last page” on the last page’s link in a bounded set. Use aria-label="last page, page [#]" to voice that the last element in the navigation is the last page in the paginated section. Don’t include this on the last navigation element in an unbounded set, since there is no last page in an unbounded set.
```

**To Reproduce**

1. Implement the pagination component
2. Notice that the last page link does not have an aria-label that tells the user it's the last page
3. Link to implementation: https://trussworks.github.io/react-uswds/?path=/docs/components-pagination--docs

**Expected behavior**
The last page link needs to have an aria-label that tells the user it's the last page.

**Screenshots**

<img width="2047" alt="Image" src="https://github.com/user-attachments/assets/3342c7bd-7ce4-4a36-ae4a-9d85442ecd58" />

**Additional context**

**Device and Browser Information (please complete the following information if describing a UI bug):**
- Mac OS
- Chrome 132.0.6834.160
